validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/MultithreadedValidatorLoader.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client.loader;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicInteger;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.infrastructure.logging.StatusLogger;
+import tech.pegasys.teku.validator.api.GraffitiProvider;
+import tech.pegasys.teku.validator.client.Validator;
+import tech.pegasys.teku.validator.client.loader.ValidatorSource.ValidatorProvider;
+
+public class MultithreadedValidatorLoader {
+
+  public static Map<BLSPublicKey, Validator> loadValidators(
+      final Map<BLSPublicKey, ValidatorProvider> providers,
+      final GraffitiProvider graffitiProvider) {
+    final int totalValidatorCount = providers.size();
+    StatusLogger.STATUS_LOG.loadingValidators(totalValidatorCount);
+
+    final ExecutorService executorService =
+        Executors.newFixedThreadPool(Math.min(4, Runtime.getRuntime().availableProcessors()));
+    try {
+      final AtomicInteger numberOfLoadedKeys = new AtomicInteger(0);
+      final List<Future<Validator>> futures =
+          providers.values().stream()
+              .map(
+                  provider ->
+                      executorService.submit(
+                          () -> {
+                            final Validator validator =
+                                new Validator(
+                                    provider.getPublicKey(),
+                                    provider.createSigner(),
+                                    graffitiProvider);
+                            int loadedValidatorCount = numberOfLoadedKeys.incrementAndGet();
+                            if (loadedValidatorCount % 10 == 0) {
```

Comment:
Would be good to log a message on the final validator as well
```suggestion
                            if (loadedValidatorCount % 10 == 0 || loadedValidatorCount == totalValidatorCount ) {
```

Reply:
Good idea.

Reply:
Turns out we had this last log message in `ValidatorLoader` to report which validators were loaded.  I've moved it into `MultithreadedValidatorLoader` so it's with the progress messages.